nano/core_test/node.cpp

```diff
@@ -2428,6 +2428,47 @@ TEST (node, unchecked_cleanup)
 	}
 }
 
+TEST (active_difficulty, recalculate_work)
+{
+	nano::system system (24000, 1);
+	auto & node1 (*system.nodes[0]);
+	nano::genesis genesis;
+	nano::keypair key1;
+	auto send1 (std::make_shared<nano::send_block> (genesis.hash (), key1.pub, 0, nano::test_genesis_key.prv, nano::test_genesis_key.pub, 0));
+	node1.work_generate_blocking (*send1);
+	uint64_t difficulty1;
+	nano::work_validate (*send1, &difficulty1);
+	// Process as local block
+	node1.process_active (send1);
+	system.deadline_set (2s);
+	while (node1.active.empty ())
+	{
+		ASSERT_NO_ERROR (system.poll ());
+	}
+	std::unique_lock<std::mutex> lock (node1.active.mutex);
+	auto existing (node1.active.roots.find (send1->qualified_root ()));
+	ASSERT_NE (node1.active.roots.end (), existing);
+	auto election (existing->election);
+	// Fake history records to force work recalculation
+	for (auto i (0); i < 20; i++)
```

Comment:
Lets test the active_difficulty numerically before and this loop in addition to testing the side effect of higher difficulty generation.